### ISSUE TITLE
Encoding without leading 0

### ISSUE THREAD

**Issue by dbrgn** [thomcc/rust-base16#3]
Would it be possible to add some kind of configuration option to omit leading zeros from hex strings?

`0f3a` would then be encoded as `f3a`.

**Comment by thomcc** [comment]
I'm not opposed to it but probably wouldn't have time to add it soon.

**Comment by dbrgn** [comment]
Fair enough! I'm currently using [this workaround](https://github.com/dbrgn/rn2xx3-rs/blob/33d8e609c40962f835458c7027530a53a6331d74/src/utils.rs#L27-L55), which works, but as an optimization, the allocation could be `nof_zero_prefix` bytes smaller.

**Comment by thomcc** [comment]
Just as a quick check, you actually mean something like omitting `000` from `000f3a`, right? Any `0x` wouldn't be added by this crate.

Assuming so, if I am reminded or remember otherwise, I'll try to do this sometime after the first week in Jan.

**Comment by dbrgn** [comment]
Yep, my original comment is a bit confusing! I'll edit.
